securedrop_client/gui/state.py

```diff
@@ -0,0 +1,49 @@
+import textwrap
+from typing import Dict, List, NewType, Optional
+from uuid import UUID
+
+from PyQt5.QtCore import QObject
+
+ConversationId = NewType("ConversationId", UUID)
+FileId = NewType("FileId", UUID)
+
+
+class State(QObject):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._selected_conversation: Optional[ConversationId] = None
+        self._conversation_files: Dict[ConversationId, List[FileId]] = {}
+
+    def selected_conversation(self) -> Optional[ConversationId]:
+        return self._selected_conversation
+
+    def conversation_files(self, id: ConversationId) -> List[FileId]:
+        return self._conversation_files.get(id, [])
+
+
+class Writer:
+    def __init__(self, state: State) -> None:
+        self._state = state
+
+    def set_selected_conversation(self, id: ConversationId) -> None:
+        if id not in self._state._conversation_files:
+            self.upsert_conversation(id, [])
+        self._state._selected_conversation = id
+        print("selected source/convesation:", id)
+
+    def upsert_conversation(self, id: ConversationId, files: List[FileId]) -> None:
+        self._state._conversation_files[id] = files
+
+
+class Preview:
+    def __init__(self, state: State) -> None:
+        self._state = state
+
+    def __repr__(self) -> str:
```

Comment:
this can be moved to the State class, and then Preview can be deleted until it does more than give a string representation of the State object

Reply:
Done.